### PR TITLE
Signed uploads

### PR DESCRIPTION
**Pull request by kafkasl**
Added signed URLs upload (list storage commit is from another PR, it will be gone after rebasing with https://github.com/Fewsats/fewsatscli/pull/4)

Related to: https://github.com/Fewsats/platform/pull/17

### CODE REVIEW

File: storage/upload.go [hunk 1]
```diff
@@ -190,8 +192,48 @@ func uploadFile(c *cli.Context) error {
 		return cli.Exit("failed to decode response", 1)
 	}
 
+	if respBody.PresignedURL != "" {
```
Comment (positiveblue) [inline storage/upload.go:200]: when could this happen?  All files will now be uploaded with the presigned url right?

File: storage/upload.go [hunk 2]
```diff
@@ -190,8 +192,48 @@ func uploadFile(c *cli.Context) error {
...
+
 	fmt.Println("File uploaded successfully.")
 	fmt.Println("Download URL: ", cfg.Domain+downloadFilePath+"/"+respBody.FileID)
 
 	return nil
 }
+
+// uploadFileToPresignedURL uploads the file to the presigned URL.
+func uploadFileToPresignedURL(presignedURL string, file []byte) error {
+	req, err := http.NewRequest(http.MethodPut, presignedURL, bytes.NewReader(file))
```
Comment (positiveblue) [inline storage/upload.go]: if we go with all field go through the presigned url and it has a limit of 5Gb, we should avoid reading the whole file in a `[]byte` as it was done until now.

We should go from `os.ReadFile(filePath)` to `os.Open(filePath)` #ILoveGolang

File: storage/upload.go (hunk 1 above)
Comment (kafkasl) [reply storage/upload.go:200]: not when using file-url upload

Comment (kafkasl) [reply storage/upload.go:200]: I've added a comment and disabled the param `file-url` for now

File: storage/upload.go (hunk 2 above)
Comment (kafkasl) [reply storage/upload.go]: done #ILoveGolang

**Pull request merged**